brigade-worker/src/k8s.ts

```diff
@@ -442,6 +442,16 @@ export class JobRunner implements jobs.JobRunner {
       }
     }
 
+    for (let cfg of job.volumeConfig) {
+      if (cfg.volume.hostPath != undefined && !project.allowHostMounts) {
+        throw new Error(`allowHostMounts is false in this project, not mounting path ${cfg.volume.hostPath.path}`)
+      }
+      this.runner.spec.volumes.push(cfg.volume);
```

Comment:
I think this should check for unique volumes?

What is the default Kubernetes behaviour if:
- a volume name is present multiple times
- a PVC is present multiple times
- a host path mount is present multiple times

Reply:
I have never got to do it myself. But I see some use with subpath, as mentioned in the issue.

Just curious, but does whether to support mounting the same volume multiple times affect our design/implementation? 🤔

Reply:
Ah I got it! You've paired a volume and its corresponding mount in each volumeConfig.

I'd rather suggest exposing `volumes` and `volumeMounts` without any abstraction first, so that we can provide a standard library feature like this `volumeConfig` on top of that, or users can build their own.

IMHO for the standard library, something that solves 80% of the use-cases will be enough, that justifies dropping support for mounting the same volume multiple times.

Reply:
Reading https://github.com/radu-matei/brigadier/commit/94f56c2446c67e817e69730ea7cf6e9f2cdd3db0 - If you agree, implementation-wise, I think adding `volumeMounts` and `volumes` that are allowed/used only when `volumeConfig` doesn't exist would suffice.

Reply:
That totally makes sense - I initially thought maybe we could simplify and boil things down to a single property, but yeah, I you're right, we should keep in line with the Kubernetes abstraction.

Updating.

Reply:
I actually don't think we should keep both `volume` + `volumeMounts` and `volumeConfig`, as it just invites confusion - what do you think?

Reply:
Btw I like the idea of pairing volumes and mounts.

I even think we can happily merge the volume and the mount as in many cases each volume is mounted only once:

```
job.volumeConfig = [
    {
        name: "modules",
        mountPath: "/lib/modules",
        readOnly: true,
        hostPath: {
            path: "/lib/modules",
            type: "Directory"
        }
    },

    {
        name: "cgroup",
        mountPath: "/sys/fs/cgroup",
        hostPath: {
            path: "/sys/fs/cgroup",
            type: "Directory"
        }
    },
    {
        name: "docker-graph-storage",
        mountPath: "/var/lib/docker",
        emptyDir: {}
    }
]
```

And I believe `name`s can be usually omitted because they can be generated from mount-path(which are guaranteed to not collide:

```
job.volumeConfig = [
    {
//        name: "lib-modules",
        mountPath: "/lib/modules",
        readOnly: true,
        hostPath: {
            path: "/lib/modules",
            type: "Directory"
        }
    },

    {
//        name: "sys-fs-cgroup",
        mountPath: "/sys/fs/cgroup",
        hostPath: {
            path: "/sys/fs/cgroup",
            type: "Directory"
        }
    },
    {
//        name: "var-lib-docker",
        mountPath: "/var/lib/docker",
        emptyDir: {}
    }
]
```

Reply:
> I actually don't think we should keep both volume + volumeMounts and volumeConfig, as it just invites confusion - what do you think?

So I'd vote for starting with only `volumes` + `volumeMounts` if I need to choose one.

// But I do believe something like `volumeConfig` has a lot of potential, and worth it, even if it needs an extensive documentation to not confuse users.

Reply:
Yeah, I initially thought that something like `volumeConfig` could be an abstraction over how volumes and volume mounts work in Kubernetes and simplify setting them up for a job.

But I agree, let's start with plain `volumes` and `volumeMounts`, then decide whether to add a higher level abstraction, how you suggested.

Thanks a lot, this has been really helpful!

Reply:
Brigadier change: https://github.com/brigadecore/brigadier/pull/22/commits/8c53a1c91bf165c69ff8205712800f120223b82b

Reply:
Updated - could you please give this another look?
Thanks!